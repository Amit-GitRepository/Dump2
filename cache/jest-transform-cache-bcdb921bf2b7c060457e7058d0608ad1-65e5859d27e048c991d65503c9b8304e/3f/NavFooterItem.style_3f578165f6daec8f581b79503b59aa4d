Object.defineProperty(exports,"__esModule",{value:true});var cov_2qq4hibf7t=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/NavFooter/NavFooterItem.style.js',hash='df7cdc5fd2e00f3b5603cb5ad0acf01afd111476',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/NavFooter/NavFooterItem.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../themes/constants.styles');exports.default=

{
wrapper:{
flexGrow:1},

container:{
alignItems:'center',
padding:5},

separator:{
borderRightWidth:2,
borderColor:_constants.colors.PRIMARY_BG_SEPARATOR},

activeTab:{
backgroundColor:_constants.colors.PRIMARY_ACTIONABLE},

tabTitle:{
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL,
fontSize:_constants.fonts.FONT_SIZE_SMALL},

activeTabTitle:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST}};