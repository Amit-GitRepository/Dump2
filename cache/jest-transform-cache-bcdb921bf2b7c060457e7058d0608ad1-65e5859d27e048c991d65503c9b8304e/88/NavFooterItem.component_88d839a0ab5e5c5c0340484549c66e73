Object.defineProperty(exports,"__esModule",{value:true});var cov_2fkx1m30jm=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/NavFooter/NavFooterItem.component.js',hash='344224dbdca970ce524dc1ab589dd8cc755ef8cc',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/NavFooter/NavFooterItem.component.js',statementMap:{'0':{start:{line:12,column:33},end:{line:14,column:3}},'1':{start:{line:13,column:4},end:{line:13,column:34}},'2':{start:{line:17,column:62},end:{line:17,column:72}},'3':{start:{line:18,column:4},end:{line:26,column:6}},'4':{start:{line:30,column:0},end:{line:37,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:12,column:21},end:{line:12,column:22}},loc:{start:{line:12,column:33},end:{line:14,column:3}},line:12},'1':{name:'(anonymous_1)',decl:{start:{line:12,column:33},end:{line:12,column:34}},loc:{start:{line:12,column:39},end:{line:14,column:3}},line:12},'2':{name:'(anonymous_2)',decl:{start:{line:16,column:2},end:{line:16,column:3}},loc:{start:{line:16,column:12},end:{line:27,column:3}},line:16}},branchMap:{'0':{loc:{start:{line:19,column:24},end:{line:19,column:92}},type:'cond-expr',locations:[{start:{line:19,column:41},end:{line:19,column:75}},{start:{line:19,column:78},end:{line:19,column:92}}],line:19},'1':{loc:{start:{line:20,column:21},end:{line:20,column:89}},type:'cond-expr',locations:[{start:{line:20,column:34},end:{line:20,column:50}},{start:{line:20,column:53},end:{line:20,column:89}}],line:20},'2':{loc:{start:{line:21,column:62},end:{line:21,column:142}},type:'cond-expr',locations:[{start:{line:21,column:79},end:{line:21,column:110}},{start:{line:21,column:113},end:{line:21,column:142}}],line:21},'3':{loc:{start:{line:22,column:37},end:{line:22,column:112}},type:'cond-expr',locations:[{start:{line:22,column:54},end:{line:22,column:94}},{start:{line:22,column:97},end:{line:22,column:112}}],line:22}},s:{'0':0,'1':0,'2':0,'3':0,'4':0},f:{'0':0,'1':0,'2':0},b:{'0':[0,0],'1':[0,0],'2':[0,0],'3':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Icon=require('../Shared/Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _ISText=require('../Shared/ISText/ISText.component');var _ISText2=_interopRequireDefault(_ISText);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _NavFooterItem=require('./NavFooterItem.style');var _NavFooterItem2=_interopRequireDefault(_NavFooterItem);
var _Touchable=require('../Shared/Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _constants=require('../../themes/constants.styles');
var _helper=require('../../language/i18n/helper');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

NavFooterItem=function(_Component){_inherits(NavFooterItem,_Component);function NavFooterItem(){var _ref;var _temp,_this,_ret;_classCallCheck(this,NavFooterItem);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=NavFooterItem.__proto__||Object.getPrototypeOf(NavFooterItem)).call.apply(_ref,[this].concat(args))),_this),_this.
navigateToScreen=function(screen){cov_2fkx1m30jm.f[0]++;cov_2fkx1m30jm.s[0]++;return function(){cov_2fkx1m30jm.f[1]++;cov_2fkx1m30jm.s[1]++;
_this.props.goToScreen(screen);
};},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(NavFooterItem,[{key:'render',value:function render()

{cov_2fkx1m30jm.f[2]++;var _ref2=(cov_2fkx1m30jm.s[2]++,
this.props),icon=_ref2.icon,screen=_ref2.screen,title=_ref2.title,isActiveScreen=_ref2.isActiveScreen,isLastItem=_ref2.isLastItem;cov_2fkx1m30jm.s[3]++;
return(
_react2.default.createElement(_Touchable2.default,{style:isActiveScreen?(cov_2fkx1m30jm.b[0][0]++,[_NavFooterItem2.default.wrapper,_NavFooterItem2.default.activeTab]):(cov_2fkx1m30jm.b[0][1]++,_NavFooterItem2.default.wrapper),onPress:this.navigateToScreen(screen)},
_react2.default.createElement(_reactNative.View,{style:isLastItem?(cov_2fkx1m30jm.b[1][0]++,_NavFooterItem2.default.container):(cov_2fkx1m30jm.b[1][1]++,[_NavFooterItem2.default.container,_NavFooterItem2.default.separator])},
_react2.default.createElement(_Icon2.default,{name:icon,size:_constants.fonts.FONT_SIZE_XXL,color:isActiveScreen?(cov_2fkx1m30jm.b[2][0]++,_constants.colors.PRIMARY_BG_TEXT_CONTRAST):(cov_2fkx1m30jm.b[2][1]++,_constants.colors.PRIMARY_TEXT_TAB_LABEL)}),
_react2.default.createElement(_ISText2.default,{type:'BOLD',style:isActiveScreen?(cov_2fkx1m30jm.b[3][0]++,[_NavFooterItem2.default.tabTitle,_NavFooterItem2.default.activeTabTitle]):(cov_2fkx1m30jm.b[3][1]++,_NavFooterItem2.default.tabTitle)},(0,_helper.translate)(title)))));




}}]);return NavFooterItem;}(_react.Component);cov_2fkx1m30jm.s[4]++;


NavFooterItem.propTypes={
icon:_propTypes2.default.string,
title:_propTypes2.default.string,
screen:_propTypes2.default.string,
goToScreen:_propTypes2.default.func,
isActiveScreen:_propTypes2.default.bool,
isLastItem:_propTypes2.default.bool};exports.default=


NavFooterItem;