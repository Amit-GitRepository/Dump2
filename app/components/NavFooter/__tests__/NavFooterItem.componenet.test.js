import NavFooterItem from '../NavFooterItem.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('NavFooterItem component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<NavFooterItem/>);
    const component = renderer.create(<NavFooterItem />).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
  it('navigateToScreen: navigate to screen to be called', () => {
    const goToScreenMock = jest.fn();
    const wrapper = shallow(<NavFooterItem goToScreen={goToScreenMock}/>);
    wrapper.instance().navigateToScreen('test')();
    expect(goToScreenMock).toHaveBeenCalledWith('test');
  });
});
