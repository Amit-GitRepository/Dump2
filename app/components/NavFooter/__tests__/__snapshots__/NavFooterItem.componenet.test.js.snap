// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavFooterItem component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "padding": 4,
          },
          Object {
            "borderColor": "#E9E9EA",
            "borderRightWidth": 2,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#303C46",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "iservice",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#303C46",
            "fontFamily": "SukhumvitSet-Bold",
            "fontSize": 14,
          }
        }
      >
        [missing "en." translation]
      </Text>
    </View>
  </View>
</View>
`;
